pyannote/audio/interactive/diarization/recipe.py

```diff
@@ -0,0 +1,430 @@
+                if distances[index_min] < 0.5:
+                    genlabel = nSpeakerVoices[index_min]["name"]
+                    if genlabel not in new_labels:
+                        new_labels.append(genlabel)
+                    for span in audio_spans:
+                        if span["label"] == label:
+                            span.update({"label": genlabel})
+
+        blocks = [{"view_id": "audio_manual"}]
+        new_labels = sorted(new_labels)
+        all_labels = new_labels + labels
+        for label in all_labels:
+            blocks.append(
+                {
+                    "view_id": "text_input",
+                    "field_id": label,
+                    "field_label": label,
+                    "field_rows": 1,
+                    "field_placeholder": label,
+                    "field_suggestions": list(nSpeakerVoices["name"]),
+                }
+            )
+
+        yield {
+            "path": path,
+            "text": text,
+            "audio": audio,
+            "audio_spans": audio_spans,
+            "audio_spans_original": deepcopy(audio_spans),
+            "chunk": {"start": excerpt.start, "end": excerpt.end},
+            "config": {"labels": all_labels, "blocks": blocks},
+            "meta": {
+                "file": filename,
+                "start": f"{excerpt.start:.1f}",
+                "end": f"{excerpt.end:.1f}",
+            },
+        }
+
+
+@prodigy.recipe(
+    "pyannote.diarization",
+    dataset=("Dataset to save annotations to", "positional", None, str),
+    source=(
+        "Path to directory containing audio files to annotate",
+        "positional",
+        None,
+        Path,
+    ),
+    pipeline=(
+        "Pretrained pipeline (name on Huggingface Hub, path to YAML file)",
+        "positional",
+        None,
+        str,
+    ),
+    chunk=(
+        "Split long audio files into shorter chunks of that many seconds each",
+        "option",
+        None,
+        float,
+    ),
+    num_classes=(
+        "Set maximum number of classes for pipelines whose number of classes is not predefined (e.g. pyannote/speaker-diarization)",
+        "option",
+        None,
+        int,
+    ),
+    embeddings=(
+        "Path to already created embeddings in a structured np.array with dtype=[('name', 'U100'), ('embedding', 'f4', dim), ('nb','i4')]",
+        "option",
+        None,
+        str,
+    ),
+    precision=("Keyboard temporal precision, in milliseconds.", "option", None, int),
+    beep=(
+        "Beep when the player reaches the end of a region.",
+        "flag",
+        None,
+        bool,
+    ),
+)
+def diarization(
+    dataset: str,
+    source: Path,
+    pipeline: Union[str, Iterable[dict]],
+    chunk: float = 30.0,
+    num_classes: int = 4,
+    embeddings: str = "",
+    precision: int = 200,
+    beep: bool = False,
+) -> Dict[str, Any]:
+
+    global nSpeakerVoices
+    pipeline = Pipeline.from_pretrained(pipeline)
```

Comment:
Which pipeline did you use in your tests? 
Shouldn't it default to `pyannote/speaker-diarization`?